File: lindera-nodejs/examples/tokenize_with_decompose.js
const { loadDictionary, Tokenizer } = require("lindera");

function main() {
  // load the dictionary
  const dictionary = loadDictionary("embedded://ipadic");

  // create a tokenizer with decompose mode
  const tokenizer = new Tokenizer(dictionary, "decompose");

  const text =
    "関西国際空港限定トートバッグを東京スカイツリーの最寄り駅であるとうきょうスカイツリー駅で買う";
  console.log(`text: ${text}\n`);

  // tokenize the text
  const tokens = tokenizer.tokenize(text);

  for (const token of tokens) {
    console.log(token.surface);
  }
}

main();

File: lindera-ruby/examples/tokenize_with_decompose.rb
# frozen_string_literal: true

require 'lindera'

# Load the dictionary
dictionary = Lindera.load_dictionary('embedded://ipadic')

# Create a tokenizer with decompose mode
tokenizer = Lindera::Tokenizer.new(dictionary, 'decompose', nil)

text = '関西国際空港限定トートバッグを東京スカイツリーの最寄り駅であるとうきょうスカイツリー駅で買う'
puts "text: #{text}\n\n"

# Tokenize the text
tokens = tokenizer.tokenize(text)

tokens.each do |token|
  puts token.surface
end
